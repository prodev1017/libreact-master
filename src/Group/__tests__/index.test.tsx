import {createElement as h} from 'react';
import {mount} from 'enzyme';
import toJson from 'enzyme-to-json';
import {Group} from '..';

describe('<Group>', () => {
  it('exists', () => {
    expect(typeof Group).toBe('function');
  });

  it('works with no children', () => {
    const wrapper = mount(<Group></Group>);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('can change element type', () => {
    const wrapper = mount(<Group as="span"></Group>);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('passes through props', () => {
    const wrapper = mount(<Group className="foo"></Group>);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('default separator is space', () => {
    const wrapper = mount(
      <Group>
        <span>Hello</span>
        <span>world</span>
      </Group>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('can set custom separator', () => {
    const wrapper = mount(
      <Group separator={<br />}>
        <span>Hello</span>
        <span>world</span>
      </Group>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});