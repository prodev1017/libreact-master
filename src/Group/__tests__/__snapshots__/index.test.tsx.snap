// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Group> can change element type 1`] = `
<Component
  as="span"
  separator=" "
>
  <span />
</Component>
`;

exports[`<Group> can set custom separator 1`] = `
<Component
  as="div"
  separator={<br />}
>
  <div>
    <span
      key=".0"
    >
      Hello
    </span>
    <br />
    <span
      key=".1"
    >
      world
    </span>
  </div>
</Component>
`;

exports[`<Group> default separator is space 1`] = `
<Component
  as="div"
  separator=" "
>
  <div>
    <span
      key=".0"
    >
      Hello
    </span>
     
    <span
      key=".1"
    >
      world
    </span>
  </div>
</Component>
`;

exports[`<Group> passes through props 1`] = `
<Component
  as="div"
  className="foo"
  separator=" "
>
  <div
    className="foo"
  />
</Component>
`;

exports[`<Group> works with no children 1`] = `
<Component
  as="div"
  separator=" "
>
  <div />
</Component>
`;
